packages/x-charts/src/BarChart/BarChart.tsx

```diff
@@ -80,9 +75,9 @@ export interface BarChartProps
    */
   axisHighlight?: ChartsAxisHighlightProps;
   /**
-   * @deprecated Consider using `slotProps.legend` instead.
+   * If `true`, the legend is not rendered.
    */
-  legend?: ChartsLegendProps;
+  hideLegend?: boolean;
```

Comment:
Nitpick: Have you discussed and agreed on this approach?
Couldn't it cause a slight confusion, when all the `legend` props are under `slotProps.legend`, but the `hidden` toggle is top level? 🤔 

P.S. On Pickers we have `tabs` and `toolbar`, which have `hidden` flags under `slotProps`.

P.P.S. Have you considered creating a `ChartsCommonProps` type to reuse `grid`, `axisHighlight`, `hideLegend`, and maybe some other props? 🤔

Reply:
I moved the hideLegend to a dedicated issue to discuss that with José when he come back #15186